homeassistant/components/smarty/sensor.py

```diff
@@ -171,12 +171,12 @@ def __init__(self, name, smarty):
                          device_class=DEVICE_CLASS_TIMESTAMP,
                          unit_of_measurement=None,
                          smarty=smarty)
-        self._days_left = 90
+        self._days_left = 91
 
     def update(self) -> None:
         """Update state."""
         _LOGGER.debug('Updating sensor %s', self._name)
         days_left = self._smarty.filter_timer
-        if days_left < self._days_left:
+        if days_left and days_left < self._days_left:
```

Comment:
When the filter is changed and the count is reset, we won't be able to show the new larger value, right?

Also, if days_left is 0 we still want to update, right?

Reply:
yes, you are right. I'll try to find a solution.

Reply:
I would remove the check on days_left and just make sure that the datetime has a constant time part that is always eg 12 PM. Then we just add the timedelta days to that datetime and that's it.

Reply:
I've just seen your comment after I've committed. Sorry about this.
Let me know what you think or I'll try to do it the way you've mentioned.